Python/sqlmlutils/packagemanagement/dependencyresolver.py

```diff
@@ -41,24 +41,23 @@ def _package_exists_on_server(self, pkgname):
                     self.clean_requirement_name(serverpkg[0].lower())
                     for serverpkg in self._server_packages])
 
+    def _check_package_meets_req(self, requirement):
+        pkgs = self._server_packages
+        req = Requirement.parse(requirement)
+        
+
     @staticmethod
     def clean_requirement_name(reqname: str):
         return reqname.replace("-", "_")
 
     @staticmethod
-    def _check_if_installed_package_meets_spec(package_tuples, name, spec):
-        op_str = spec[0]
-        req_version = spec[1]
-
+    def _check_if_installed_package_meets_spec(package_tuples, requirement):
         installed_package_name_and_version = [package for package in package_tuples \
-            if DependencyResolver.clean_requirement_name(name.lower()) == \
+            if DependencyResolver.clean_requirement_name(requirement.name.lower()) == \
                 DependencyResolver.clean_requirement_name(package[0].lower())]
             
         if not installed_package_name_and_version:
             return False
 
-        installed_package_name_and_version = installed_package_name_and_version[0]
-        installed_version = installed_package_name_and_version[1]
-
-        operator_map = {'>': 'gt', '>=': 'ge', '<': 'lt', '==': 'eq', '<=': 'le', '!=': 'ne'}
-        return getattr(operator, operator_map[op_str])(LooseVersion(installed_version), LooseVersion(req_version))
+        installed_version = installed_package_name_and_version[0][1]

```

Comment:
installed_package_name_and_version is a list of tuples of packages (name,version). It is actually a list of size 1 since we find which exact package in package_tuples matches our requirement.

So here we get the version of that package. [0] accesses the first element of the list and [1] gets the version from the tuple.